### ISSUE TITLE
The visibility AABB of CPUParticles2D cannot be adjusted

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot#23960]
**Godot version:** Git https://github.com/godotengine/godot/commit/8348aca118311b82f632781b19230471fae56d4a

**OS/device including version:** Fedora 29, NVIDIA 410.73 (running in GLES3)

**Issue description:**

The visibility AABB of CPUParticles2D cannot be adjusted, which means that particles may be culled too early when they are parented to moving nodes (or when the camera moves away from them). According to @reduz, the visibility AABB should be generated on the fly.

**Steps to reproduce:**

Attach a CPUParticles2D node with Local Coords *disabled* to a moving parent (to create a particle trail) and wait until the parent leaves the screen.

**Minimal reproduction project:** [Calinou/escape-space](https://github.com/Calinou/escape-space) before commit https://github.com/Calinou/escape-space/commit/20987e6726ad6a7c6916d1638d5b00b35636e84a (run the project, hit some balls, wait until they leave the screen)

**Comment by eon-s** [comment]
This is marked to solve on 3.2 but is a bug that makes CPUParticles2D useless, would be good if the visibility rectangle can be changed by code at least.

**Comment by eon-s** [comment]
Found a workaround for the time being:
On particles _ready do:
```gdscript
VisualServer.canvas_item_set_custom_rect(get_canvas_item(),true,your_visibility_rect)
```

**Comment by arnoldochavez** [comment]
I was facing the same issue, then just set preprocess to 1 in the CPUParticle2D node and now it doesn't get culled out when outside of the screen

**Comment by JFonS** [comment]
Should be fixed by #25983.
